### DISCUSSION TITLE
Does Intellij Rust has the ability to turn off specific macro expansion?

### DISCUSSION THREAD

**Discussion by xzbdmw** [intellij-rust/intellij-rust#10783] [Q&A]
Similar to this
<img width="1116" alt="image" src="https://github.com/intellij-rust/intellij-rust/assets/97848247/2620d66f-96bb-4253-8f54-6b150bd72025">

**Comment by vlad20012** [comment]
There isn't a configuration option for this, but we have a hardcoded list of ignored macros. The list includes `tokio::main`, for example

**Reply by sebadob** [reply]
Would it be possible to include the `server` and `component` macro from [leptos](https://github.com/leptos-rs/leptos) here as well?  
This is even suggested for a better DX (https://leptos-rs.github.io/leptos/appendix_dx.html).

When these macros are expanded, the IDE gets extremely slow and sometimes something under the hood even crashes. Can be fixed with out-commenting the macro for the whole function and then turn it back on. However, this is really annoying.

**Reply by sebadob** [reply]
I have never taken a look at the code of the `intellij-rust` plugin before, but I opened a small PR which should achieve exactly this (I guess). I tested some problems with the dev IDE started from this PR and it seems to work.

You can find it here:  https://github.com/intellij-rust/intellij-rust/pull/10873
